Add changelog for KKP 2.28 (#14712)
kubermatic/kubermatic: docs/changelogs/CHANGELOG-2.28.md

# Kubermatic 2.28

- [v2.28.0](#v2280)

## v2.28.0

**GitHub release: [v2.28.0](https://github.com/kubermatic/kubermatic/releases/tag/v2.28.0)**

Before upgrading, make sure to read the [general upgrade guidelines](https://docs.kubermatic.com/kubermatic/v2.28/installation/upgrading/). Consider tweaking `seedControllerManager.maximumParallelReconciles` to ensure user cluster reconciliations will not cause resource exhaustion on seed clusters. A [full upgrade guide is available from the official documentation](https://docs.kubermatic.com/kubermatic/v2.28/installation/upgrading/upgrade-from-2.27-to-2.28/).

### Breaking Changes

> This release contains changes that require additional attention, please read the following items carefully.

Reviewer: Remove this line.. the section name is self explanatory.

> - The deprecated Oauth helm chart has been removed. Please follow https://docs.kubermatic.com/kubermatic/v2.27/installation/upgrading/upgrade-from-2.26-to-2.27/#dex-v242 to migrate from Oauth to the new upstream Dex chart ([#14657](https://github.com/kubermatic/kubermatic/pull/14657))

Reviewer: Replace 
`- The deprecated OAuth helm chart has been removed. Before proceeding with KKP upgrade, please follow the [dex migration procedure](https://docs.kubermatic.com/kubermatic/v2.27/installation/upgrading/upgrade-from-2.26-to-2.27/#dex-v242) to migrate from oauth/dex to the new upstream Dex chart ([#14657](https://github.com/kubermatic/kubermatic/pull/14657))`

> - `deploy-default-app-catalog` for kubermatic-installer has been deprecated and would have no affect going forward. The field `.Spec.Applications.DefaultApplicationCatalog` can be used instead to manage default application catalog ([#14697](https://github.com/kubermatic/kubermatic/pull/14697))

Reviewer: Replace with 
```- `deploy-default-app-catalog` option from kubermatic-installer has been deprecated and would have no affect going forward. The field `.Spec.Applications.DefaultApplicationCatalog` can be used instead to manage default application catalog ([#14697](https://github.com/kubermatic/kubermatic/pull/14697))```

- Support for multiple security groups has been dropped for OpenStack. If you are using multiple security groups for an OpenStack cluster, you need to manually adjust the `cluster.Spec.Cloud.Openstack.SecurityGroups` ([#14269](https://github.com/kubermatic/kubermatic/pull/14269))

> - Node-exporter chart is now using the upstream helm chart. This means there are some adjustment required to the `values.yaml`. replace the `nodeExporter` with `node-exporter` ([#14176](https://github.com/kubermatic/kubermatic/pull/14176))

Reviewer: include other node-exporter PR ([#14670](https://github.com/kubermatic/kubermatic/pull/14670)) to the PR list.  
And remove entry from https://github.com/mohamed-rafraf/kubermatic/blob/changelog-2.28/docs/changelogs/CHANGELOG-2.28.md?plain=1#L78

> - Alertmanager chart was replaced with upstream chart. This means, some of the old values might need migration. Please review documentation ([#14175](https://github.com/kubermatic/kubermatic/pull/14175))

Reviewer: Similar change here. Add https://github.com/mohamed-rafraf/kubermatic/blob/changelog-2.28/docs/changelogs/CHANGELOG-2.28.md?plain=1#L77 this to the PR list.

### ACTION REQUIRED

- Update cert-manager to v1.16.5. So the following updates should be done values.yaml ([#14400](https://github.com/kubermatic/kubermatic/pull/14400)): 
    - update `webhook.replicas` to `webhook.replicaCount` 
    - update `cainjector.replicas` to `webhook.replicaCount` 
    - remove `webhook.injectAPIServerCA` 

> - Blackbox-exporter chart is now using the upstream helm chart indirectly. So the customizations for blackbox-exporter via helm values.yaml should be moved under key `blackbox-exporter` instead of `blackboxExporter` ([#14170](https://github.com/kubermatic/kubermatic/pull/14170))

Reviewer: This should be part of `Breaking Changes`

### New Features

> - LocalAI was added to the default application catalog for the enterprise edition ([#14700](https://github.com/kubermatic/kubermatic/pull/14700))

Reviewer: Replace with 
`- EE: LocalAI is added to the default applications catalog ([#14700](https://github.com/kubermatic/kubermatic/pull/14700))`

- The default apps from the enterprise catalog are now also part of the kubermatic-installer `mirror-images` command ([#14683](https://github.com/kubermatic/kubermatic/pull/14683))
- Add Kyverno images to `mirror-images` command ([#14681](https://github.com/kubermatic/kubermatic/pull/14681))
- Add `--insecure` flag to mirror-images command to bypass TLS verification ([#14635](https://github.com/kubermatic/kubermatic/pull/14635))

> - Add support for configuring a Authorization Webhook for UserClusters ([#13930](https://github.com/kubermatic/kubermatic/pull/13930))

Reviewer: Replace with 
`- Add support for configuring an Authorization Webhook for the User Clusters ([#13930](https://github.com/kubermatic/kubermatic/pull/13930))`

> - Mla-secrets: chart can now be safely redeployed, existing secrets take precedence over provided values ([#14568](https://github.com/kubermatic/kubermatic/pull/14568))

Reviewer: Replace with 
`- Mla-secrets chart can now be safely redeployed, existing secrets take precedence over provided values ([#14568](https://github.com/kubermatic/kubermatic/pull/14568))`

> - Gateway API CRDs are now automatically installed in user clusters when KubeLB integration is enabled ([#14368](https://github.com/kubermatic/kubermatic/pull/14368))

Reviewer: Replace with 
`- Gateway API CRDs are now automatically installed in the user clusters, when KubeLB integration is enabled ([#14368](https://github.com/kubermatic/kubermatic/pull/14368))`

- Add support for scheduling options (tolerations, affinity, nodeSelector) to the KKP operator Helm chart, allowing users to control pod placement in their clusters ([#14574](https://github.com/kubermatic/kubermatic/pull/14574))
- Add nginx config to increase header size for Dashboard and Dex ([#14579](https://github.com/kubermatic/kubermatic/pull/14579))

> - The default Policy Template Catalog can be deployed via --deploy-default-policy-template-catalog flag (EE) ([#14472](https://github.com/kubermatic/kubermatic/pull/14472))

Reviewer: Replace with 
```- EE: The default Policy Template Catalog can be deployed via `--deploy-default-policy-template-catalog` flag  ([#14472](https://github.com/kubermatic/kubermatic/pull/14472))```

> - A new --limit-apps flag has been added to the Kubermatic Installer, allowing users to limit which AppDefinitions are installed during the setup process. This flag accepts a comma-separated list of AppDefinition names. If the flag is not provided or the list is empty, all available AppDefinitions will be installed—provided the default app catalog is enabled ([#14569](https://github.com/kubermatic/kubermatic/pull/14569))

Reviewer: Should be 
```- A new `--limit-apps` flag has been added to the Kubermatic Installer, allowing users to limit which AppDefinitions are installed during the setup process. This flag accepts a comma-separated list of AppDefinition names. If the flag is not provided or the list is empty, all available AppDefinitions will be installed—provided the default app catalog is enabled ([#14569](https://github.com/kubermatic/kubermatic/pull/14569)```

- Fix `--skip-seed-validation` flag on the KKP installer ([#14585](https://github.com/kubermatic/kubermatic/pull/14585))
- Correctly mounts the custom CA bundle ConfigMap to fix reconciliation failures in custom CA environments ([#14575](https://github.com/kubermatic/kubermatic/pull/14575))
- Fix a bug where CSI Snapshot validating webhook was being deployed even if the CSI drivers are disabled for a cluster. When the csi driver is disabled after cluster creation the both mentioned resources will be cleaned up now ([#14466](https://github.com/kubermatic/kubermatic/pull/14466))
- Remove old warnings for new dex chart ([#14423](https://github.com/kubermatic/kubermatic/pull/14423))
- Fix the service account deletion process ([#14371](https://github.com/kubermatic/kubermatic/pull/14371))
- Ensure that etcd backup images are pulled from the overwrite Registry in air-gapped environments ([#14356](https://github.com/kubermatic/kubermatic/pull/14356))
- Fix a bug for KubeLB where disabling the ingress class for a user cluster was not working ([#14396](https://github.com/kubermatic/kubermatic/pull/14396))

> - Kube-state-metrics chart is now using the upstream helm chart with app version 2.15.0 ([#14357](https://github.com/kubermatic/kubermatic/pull/14357))

Reviewer: This should be moved to breaking change. and include comma separate other PRs related kube-state-metrics from the changelog.

### ACTION REQUIRED

- Update cert-manager to v1.16.5. So the following updates should be done values.yaml ([#14400](https://github.com/kubermatic/kubermatic/pull/14400)): 
    - update `webhook.replicas` to `webhook.replicaCount` 
    - update `cainjector.replicas` to `webhook.replicaCount` 
    - remove `webhook.injectAPIServerCA` 

### API Changes

- The KKP APIs have been moved into a Go module named `k8c.io/kubermatic/v2/sdk` ([#14171](https://github.com/kubermatic/kubermatic/pull/14171))

### Supported Kubernetes Versions

- Remove support for Kubernetes 1.29 ([#14345](https://github.com/kubermatic/kubermatic/pull/14345))
- Add support for Kubernetes 1.33 ([#14419](https://github.com/kubermatic/kubermatic/pull/14419))
- Add 1.32.4/1.32.3/1.31.8/1.31.7/1.30.12/1.30.11 to the list of supported Kubernetes releases ([#14266](https://github.com/kubermatic/kubermatic/pull/14266)) ([#14385](https://github.com/kubermatic/kubermatic/pull/14385))

#### Supported Versions
* 1.30.11
* 1.30.12
* 1.31.7
* 1.31.8
* 1.32.3
* 1.32.4
* 1.33.0

### Cloud Providers 
#### KubeVirt
- Support KubeVirt Subnet CIDR in the Seed object ([#14604](https://github.com/kubermatic/kubermatic/pull/14604))

> - The field `enabledDedicatedCPUs` in kubevirt provider spec is now deprecated. A new field called `useDomainCPU` is introduced which is for the same purpose. When set to `true` cpu will be assigned by `spec.domain.cpu` for a kubevirt virtual machine instead of using resource requests and limits ([#14621](https://github.com/kubermatic/kubermatic/pull/14621))

Reviewer: ```suggestion
- The field `enabledDedicatedCPUs` in KubeVirt provider spec is now deprecated. A new field called `useDomainCPU` is introduced which is for the same purpose. When set to `true` cpu will be assigned by `spec.domain.cpu` for a KubeVirt virtual machine instead of using resource requests and limits ([#14621](https://github.com/kubermatic/kubermatic/pull/14621))
```

- Update KubeVirt CSI Driver Operator to v0.4.3 ([#14178](https://github.com/kubermatic/kubermatic/pull/14178))

> - A bug that caused network policies to not be removed from the kubevirt infra cluster has been fixed ([#14429](https://github.com/kubermatic/kubermatic/pull/14429))

Reviewer: ```suggestion
- A bug that caused network policies to not be removed from the KubeVirt infra cluster has been fixed ([#14429](https://github.com/kubermatic/kubermatic/pull/14429))
```

- Support `infra-csi-driver` as a `volumeProvisioner` for the KubeVirt CSI Driver ([#14199](https://github.com/kubermatic/kubermatic/pull/14199))

> - Add ability to disable automatic installation of default kubevirt instance types and preferences ([#14426](https://github.com/kubermatic/kubermatic/pull/14426))

Reviewer: ```suggestion
- Add ability to disable automatic installation of default KubeVirt instance types and preferences ([#14426](https://github.com/kubermatic/kubermatic/pull/14426))
```

> - KubeLB: new field `extraArgs` has been introduced for KubeLB at the seed and cluster level. This field can be used to configure extra arguments for the kubeLB CCM ([#14564](https://github.com/kubermatic/kubermatic/pull/14564))

Reviewer: Update as follows
`- KubeLB: New field `extraArgs` has been introduced for KubeLB at the Seed and cluster level. This field can be used to configure extra arguments for the KubeLB CCM ([#14564](https://github.com/kubermatic/kubermatic/pull/14564))`

> - KubeLB: the configuration in seed `seed.spec.kubelb.enableForAllDatacenters` can be used to allow kubeLB installation for all the datacenters belonging to the seed ([#14558](https://github.com/kubermatic/kubermatic/pull/14558))

Reviewer: Update as follows
`- KubeLB: The configuration in the Seed `seed.spec.kubelb.enableForAllDatacenters` can be used to allow KubeLB installation for all the datacenters belonging to the Seed ([#14558](https://github.com/kubermatic/kubermatic/pull/14558))`

> - A new field `spec.datacenters.<example-dc>.spec.kubevirt.enableDedicatedCpus` was added to seed crd to control whether kubevirt machine cpus are configured by `spec.template.spec.domain.resources` with requests and limits or `spec.template.spec.domain.cpu` . Later one is required to use kubevirt cpu allocation ratio feature ([#14298](https://github.com/kubermatic/kubermatic/pull/14298))

Reviewer: ```suggestion
- A new field `spec.datacenters.<example-dc>.spec.kubevirt.enableDedicatedCpus` was added to Seed CRD to control whether KubeVirt machine CPUs are configured by `spec.template.spec.domain.resources` with requests and limits or `spec.template.spec.domain.cpu` . Later one is required to use KubeVirt CPU allocation ratio feature ([#14298](https://github.com/kubermatic/kubermatic/pull/14298))
```

> - Add support for tagging cluster backup objects in Kubermatic for improved management and traceability. It can help to identify, categorise, and track backup resources more effectively across multiple clusters and tenants ([#14373](https://github.com/kubermatic/kubermatic/pull/14373))

Reviewer: Instead of `in Kubermatic` update to `in KKP`

> - A new field was introduced for kubevirt provider in namespacedmode in enterprise edition to configure a mode for the deployed network policy in kubevirt infrastructure clusters. Default mode is `allow` which allows by default all traffic except to other providernetwork subnets. The other option is `deny` which denies all traffic except to the usercluster apiserver, configured nameservers and between worker nodes ([#14390](https://github.com/kubermatic/kubermatic/pull/14390))

Reviewer: ```suggestion
- EE: A new field was introduced for KubeVirt provider in namespaced mode to configure a mode for the deployed network policy in KubeVirt infrastructure clusters. Default mode is `allow` which allows by default all traffic except to other provider network subnets. The other option is `deny` which denies all traffic except to the user cluster apiserver, configured nameservers and between worker nodes ([#14390](https://github.com/kubermatic/kubermatic/pull/14390))
```

- Add support for using a Proxy Between KKP and the nodeport-proxy ([#14159](https://github.com/kubermatic/kubermatic/pull/14159))
- Add a new field in KubermaticSettings to allow setting a default checksum algorithm for Velero through Dashboard ([#14253](https://github.com/kubermatic/kubermatic/pull/14253))
- Introduce the `mirror-binaries` in `kubermatic-installer` to mirror the kubernetes and container tools binaries for Offline setups ([#14251](https://github.com/kubermatic/kubermatic/pull/14251))
- Ensure `mirror-images` processes all images without blocking, logging failed images at the end for better visibility and debugging ([#14262](https://github.com/kubermatic/kubermatic/pull/14262))
- Include `cilium-envoy` image in the mirrored images ([#14238](https://github.com/kubermatic/kubermatic/pull/14238))
- Add a new field in `Cluster` to configure HTTP Proxy at User Cluster Level ([#14209](https://github.com/kubermatic/kubermatic/pull/14209))

### API Changes

- The KKP APIs have been moved into a Go module named `k8c.io/kubermatic/v2/sdk` ([#14171](https://github.com/kubermatic/kubermatic/pull/14171))

> ### Deprecations

Reviewer: Deprecations should be added after Cleanup section

- Fix a Go panic when using git-source in Applications ([#14219](https://github.com/kubermatic/kubermatic/pull/14219))
- Include the etcd backup restore and delete images in the kubermatic-installer mirror-images command ([#14220](https://github.com/kubermatic/kubermatic/pull/14220))
- Add dex and gitops charts to the CI release pipeline for inclusion in the release tar ([#14192](https://github.com/kubermatic/kubermatic/pull/14192))
- Fix a bug that prevents configuring `resources` in KNP deployments ([#14205](https://github.com/kubermatic/kubermatic/pull/14205))
- Apply override registry configuration to cilium-envoy images ([#14164](https://github.com/kubermatic/kubermatic/pull/14164))
- The local kind command from the kubermatic-installer is now using helm values to deploy dex by the upstream helm chart. This was required due to removal of the old custom chart. ([#14704](https://github.com/kubermatic/kubermatic/pull/14704))

### Cleanups

- Cluster-autoscaler has been removed from the default accessible addons list ([#14689](https://github.com/kubermatic/kubermatic/pull/14689))
- Deprecate Equinix Metal provider ([#14448](https://github.com/kubermatic/kubermatic/pull/14448))

> ### Supported Kubernetes Versions

Reviewer: Move this above after `API Changes` section

- Remove support for Kubernetes 1.29 ([#14345](https://github.com/kubermatic/kubermatic/pull/14345))
- Add support for Kubernetes 1.33 ([#14419](https://github.com/kubermatic/kubermatic/pull/14419))
- Add 1.32.4/1.32.3/1.31.8/1.31.7/1.30.12/10.30.11 to the list of supported Kubernetes releases ([#14266](https://github.com/kubermatic/kubermatic/pull/14266)) ([#14385](https://github.com/kubermatic/kubermatic/pull/14385))

> ### Cloud Providers 

Reviewer: Similarly move this up should be before `New Features`

- Default Application Catalog can now be managed via KubermaticConfiguration through the field `.Spec.Applications.DefaultApplicationCatalog` ([#14697](https://github.com/kubermatic/kubermatic/pull/14697))

### Miscellaneous

- The deprecated k8sgpt application has been removed and was replaced by the k8sgpt-operator app instead ([#14403](https://github.com/kubermatic/kubermatic/pull/14403))
- Disable cilium-envoy daemonset, if it was not specified in the chart values ([#14173](https://github.com/kubermatic/kubermatic/pull/14173))

### Dashboard and API 

#### Breaking Changes

- The `usePodResourcesCPU` feature will replace `enableDedicatedCPUs` flag. In the time of deprecation both are taking effect but the new value will have more priority. When `enableDedicatedCPUs` is set to `false` which is also the default value, you need to set `usePodResourcesCPU` to `true` to keep the same behaviour as before for new created machines. If `enableDedicatedCPUs` was set to `true` nothing needs to be changed ([#7413](https://github.com/kubermatic/dashboard/pull/7413))

#### Cloud Providers

##### KubeVirt

> - Add the ability to disable the automatic installation of default kubevirt instance types and preferences ([#7304](https://github.com/kubermatic/dashboard/pull/7304))

Reviewer: ```suggestion
- Add the ability to disable the automatic installation of default KubeVirt instance types and preferences ([#7304](https://github.com/kubermatic/dashboard/pull/7304))
```

- Display KubeVirt Subnet CIDRs in UI ([#7369](https://github.com/kubermatic/dashboard/pull/7369))

> - The kkp api is now aware on how to configure cpus for kubevirt virtual machines based on a new introduced field in kkp seed crd called `spec.datacenters.<example-dc>.spec.kubevirt.enableDedicatedCpus` ([#7252](https://github.com/kubermatic/dashboard/pull/7252))

Reviewer: ```suggestion
- The KKP API is now aware on how to configure CPUs for KubeVirt virtual machines based on a new introduced field in KKP Seed CRD called `spec.datacenters.<example-dc>.spec.kubevirt.enableDedicatedCpus` ([#7252](https://github.com/kubermatic/dashboard/pull/7252))
```

- Node-exporter chart is now using the upstream helm chart. This means there are some adjustment required to the `values.yaml`. replace the `nodeExporter` with `node-exporter` ([#14176](https://github.com/kubermatic/kubermatic/pull/14176)) ([#14670](https://github.com/kubermatic/kubermatic/pull/14670))
- Alertmanager chart was replaced with upstream chart. This means, some of the old values might need migration. Please review documentation ([#14175](https://github.com/kubermatic/kubermatic/pull/14175))
- Blackbox-exporter chart is now using the upstream helm chart indirectly. So the customizations for blackbox-exporter via helm values.yaml should be moved under key `blackbox-exporter` instead of `blackboxExporter` ([#14170](https://github.com/kubermatic/kubermatic/pull/14170))
- Kube-state-metrics chart is now using the upstream helm chart with app version 2.15.0 ([#14357](https://github.com/kubermatic/kubermatic/pull/14357))

> 

Reviewer 2: ```suggestion
- New roles for the Kubevirt csi driver have been added and users must add these role to KubeVirt infra cluster kubeconfig service account (https://github.com/kubermatic/kubermatic/pull/14640/files#diff-3f5838d7684c2eb21e0f5e55484827bdf8144c66695d8421a727b66db2abec71R72-R81)
```

### ACTION REQUIRED

- Update cert-manager to v1.16.5. So the following updates should be done values.yaml ([#14400](https://github.com/kubermatic/kubermatic/pull/14400)): 
    - update `webhook.replicas` to `webhook.replicaCount` 
    - update `cainjector.replicas` to `webhook.replicaCount` 
    - remove `webhook.injectAPIServerCA` 

- Kube-state-metrics chart is now using the upstream helm chart. This means there are some adjustments required to the `values.yaml`. The name needs to be overridden. 
    ```yaml
    kube-state-metrics:
    fullnameOverride: kube-state-metrics
    ```

### Supported Kubernetes Versions

- Remove support for Kubernetes 1.29 ([#14345](https://github.com/kubermatic/kubermatic/pull/14345))
- Add support for Kubernetes 1.33 ([#14419](https://github.com/kubermatic/kubermatic/pull/14419))

> - Add 1.32.4/1.32.3/1.31.8/1.31.7/1.30.12/10.30.11 to the list of supported Kubernetes releases ([#14266](https://github.com/kubermatic/kubermatic/pull/14266)) ([#14385](https://github.com/kubermatic/kubermatic/pull/14385))

Reviewer: Fix typo `10.30.11` to `1.30.11`

> 

Reviewer: Add sub section 

`#### Supported Versions

- 1.30.11
- 1.30.12
- 1.31.7
- 1.31.8
- 1.32.3
- 1.32.4
- 1.33.0
`

- A new field `spec.datacenters.<example-dc>.spec.kubevirt.enableDedicatedCpus` was added to seed crd to control whether kubevirt machine cpus are configured by `spec.template.spec.domain.resources` with requests and limits or `spec.template.spec.domain.cpu` . Later one is required to use kubevirt cpu allocation ratio feature ([#14298](https://github.com/kubermatic/kubermatic/pull/14298))
- A new field was introduced for kubevirt provider in namespacedmode in enterprise edition to configure a mode for the deployed network policy in kubevirt infrastructure clusters. Default mode is `allow` which allows by default all traffic except to other providernetwork subnets. The other option is `deny` which denies all traffic except to the usercluster apiserver, configured nameservers and between worker nodes ([#14390](https://github.com/kubermatic/kubermatic/pull/14390))
- Support KubeVirt CCM Load Balancer Interface Disabling ([#14521](https://github.com/kubermatic/kubermatic/pull/14521))

#### OpenStack
- Fix reconciliation loop for routers in Openstack provider by allowing updates to the `routerID` field ([#14420](https://github.com/kubermatic/kubermatic/pull/14420))

> - User-clusters in Openstack can share a router, which is deleted only after all associated clusters are removed ([#14468](https://github.com/kubermatic/kubermatic/pull/14468))

Reviewer: Update `User-clusters` to `User clusters`

- KubeLB: New field extraArgs has been introduced for KubeLB at the Seed and cluster level. This field can be used to configure extra arguments for the KubeLB CCM ([#14564](https://github.com/kubermatic/kubermatic/pull/14564))
- KubeLB: The configuration in the Seed seed.spec.kubelb.enableForAllDatacenters can be used to allow KubeLB installation for all the datacenters belonging to the Seed ([#14558](https://github.com/kubermatic/kubermatic/pull/14558))
- Install cluster-autoscaler by default as a system application via kubermatic-operator ([#14509](https://github.com/kubermatic/kubermatic/pull/14509))
- Add support for tagging cluster backup objects in KKP for improved management and traceability. It can help to identify, categorise, and track backup resources more effectively across multiple clusters and tenants ([#14373](https://github.com/kubermatic/kubermatic/pull/14373))
- Add `AuditLogging` configuration on seed level ([#14464](https://github.com/kubermatic/kubermatic/pull/14464))

> - Add the ability to define the allowed IP Ranges for the API server for the user-cluster on the Seed Level ([#14462](https://github.com/kubermatic/kubermatic/pull/14462))

Reviewer: remove `-` from `user-cluster`, ie. update to `user cluster`

- Node-exporter chart is now using the upstream helm chart. This means there are some adjustment required to the `values.yaml`. replace the `nodeExporter` with `node-exporter` ([#14176](https://github.com/kubermatic/kubermatic/pull/14176)) ([#14670](https://github.com/kubermatic/kubermatic/pull/14670))
- Alertmanager chart was replaced with upstream chart. This means, some of the old values might need migration. Please review documentation ([#14175](https://github.com/kubermatic/kubermatic/pull/14175))  ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))
- Blackbox-exporter chart is now using the upstream helm chart indirectly. So the customizations for blackbox-exporter via helm values.yaml should be moved under key `blackbox-exporter` instead of `blackboxExporter` ([#14170](https://github.com/kubermatic/kubermatic/pull/14170))  ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))
- Kube-state-metrics chart is now using the upstream helm chart with app version 2.15.0 ([#14357](https://github.com/kubermatic/kubermatic/pull/14357)) ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))

> - New roles for the Kubevirt csi driver have been added and users must add these role to KubeVirt infra cluster kubeconfig service account [#14502]( https://github.com/kubermatic/kubermatic/pull/14502/files#diff-3f5838d7684c2eb21e0f5e55484827bdf8144c66695d8421a727b66db2abec71R72-R81)

Reviewer 2: ```suggestion
- New roles for the Kubevirt csi driver have been added and users must add these role to KubeVirt infra cluster kubeconfig service account ([#14502]( https://github.com/kubermatic/kubermatic/pull/14502))
```

- Node-exporter chart is now using the upstream helm chart. This means there are some adjustment required to the `values.yaml`. replace the `nodeExporter` with `node-exporter` ([#14176](https://github.com/kubermatic/kubermatic/pull/14176)) ([#14670](https://github.com/kubermatic/kubermatic/pull/14670))
- Alertmanager chart was replaced with upstream chart. This means, some of the old values might need migration. Please review documentation ([#14175](https://github.com/kubermatic/kubermatic/pull/14175))  ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))
- Blackbox-exporter chart is now using the upstream helm chart indirectly. So the customizations for blackbox-exporter via helm values.yaml should be moved under key `blackbox-exporter` instead of `blackboxExporter` ([#14170](https://github.com/kubermatic/kubermatic/pull/14170))  ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))
- Kube-state-metrics chart is now using the upstream helm chart with app version 2.15.0 ([#14357](https://github.com/kubermatic/kubermatic/pull/14357)) ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))

> - New roles for the Kubevirt csi driver have been added and users must add these role to KubeVirt infra cluster kubeconfig service account (https://github.com/kubermatic/kubermatic/pull/14640/files#diff-3f5838d7684c2eb21e0f5e55484827bdf8144c66695d8421a727b66db2abec71R72-R81)

Reviewer 2: ```suggestion
- New roles for the Kubevirt csi driver have been added and users must add these role to KubeVirt infra cluster kubeconfig service account [#14502]
```

- Support disabling default OperatingSystemProfiles in user clusters ([#14515](https://github.com/kubermatic/kubermatic/pull/14515))
- Add a new optional field called `args` under KonnectivityProxySettings to allow users to specify a set of arguments for Konnectivity deployments ([#14189](https://github.com/kubermatic/kubermatic/pull/14189))

### API Changes

- The KKP APIs have been moved into a Go module named `k8c.io/kubermatic/v2/sdk` ([#14171](https://github.com/kubermatic/kubermatic/pull/14171))

### Bugfixes

- Fix `kubermatic-installer local kind` command for EE setups to set correct image pull secrets in `values.yaml` ([#14707](https://github.com/kubermatic/kubermatic/pull/14707))
- A bug was fixed where enforced annotation on application installation were not removed when enforcement for the related application definition was disabled ([#14706](https://github.com/kubermatic/kubermatic/pull/14706))
- Add the ability to skip charts in the `kubermatic-installer deploy usercluster-mla` command ([#14688](https://github.com/kubermatic/kubermatic/pull/14688))

> - Fix upgrade process for Seed MLA components: 

Reviewer: This entry should be removed as moved under Breaking Changes.

    * Alertmanager chart is now using the upstream helm chart with app version v0.28.0 
    * kube-state-metrics chart is now using the upstream helm chart with app version 2.15.0 
    * blackbox-exporter chart is now using the upstream helm chart with app version v0.25.0 ([#14675](https://github.com/kubermatic/kubermatic/pull/14675))

> - Node-exporter chart is now using the upstream helm chart with app version 1.19.0 ([#14670](https://github.com/kubermatic/kubermatic/pull/14670))

Reviewer: This entry should be removed as well as move to breaking changes.

- Fix a Go panic when using git-source in Applications ([#14219](https://github.com/kubermatic/kubermatic/pull/14219))
- Include the etcd backup restore and delete images in the kubermatic-installer mirror-images command ([#14220](https://github.com/kubermatic/kubermatic/pull/14220))
- Add dex and gitops charts to the CI release pipeline for inclusion in the release tar ([#14192](https://github.com/kubermatic/kubermatic/pull/14192))
- Fix a bug that prevents configuring `resources` in KNP deployments ([#14205](https://github.com/kubermatic/kubermatic/pull/14205))
- Apply override registry configuration to cilium-envoy images ([#14164](https://github.com/kubermatic/kubermatic/pull/14164))
- The local kind command from the kubermatic-installer is now using helm values to deploy dex by the upstream helm chart. This was required due to removal of the old custom chart. ([#14704](https://github.com/kubermatic/kubermatic/pull/14704))

### Miscellaneous

> - A bug where images with digest instead of a tag are not properly parsed was fixed. This affected mirroring images and all parsing with an overwrite registry configured ([#14664](https://github.com/kubermatic/kubermatic/pull/14664))

Reviewer: Move this under Bugfixes section

> - KubeLB: CCM will adjust the tenant kubeconfig to use API server endpoint and CA certificate from the management kubeconfig that is provided to KKP at the seed/datacenter level ([#14522](https://github.com/kubermatic/kubermatic/pull/14522))

Reviewer: Move this under `Bugfixes

> - Support disabling default OperatingSystemProfiles in user clusters ([#14515](https://github.com/kubermatic/kubermatic/pull/14515))

Reviewer: Move this under `New Features`

> - The deprecated k8sgpt application has been removed and was replaced by the k8sgpt-operator app instead ([#14403](https://github.com/kubermatic/kubermatic/pull/14403))

Reviewer: Move this under Deprecations section

> - Add a new optional field called `args` under KonnectivityProxySettings to allow users to specify a set of arguments for Konnectivity deployments ([#14189](https://github.com/kubermatic/kubermatic/pull/14189))

Reviewer: Move this under `New Features`

- Support MatchSubnetAndStorageLocation and Subnets Regions and Zones ([#14414](https://github.com/kubermatic/kubermatic/pull/14414))
- Kube-state-metrics chart is now using the upstream helm chart with app version 2.15.0 ([#14174](https://github.com/kubermatic/kubermatic/pull/14174))

### Cleanups

- Cluster-autoscaler has been removed from the default accessible addons list ([#14689](https://github.com/kubermatic/kubermatic/pull/14689))
- Deprecate Equinix Metal provider ([#14448](https://github.com/kubermatic/kubermatic/pull/14448))

### Deprecations

- `deploy-default-app-catalog` for kubermatic-installer has been deprecated and would have no affect going forward. The field `.Spec.Applications.DefaultApplicationCatalog` can be used instead to manage default application catalog ([#14697](https://github.com/kubermatic/kubermatic/pull/14697))
- Default Application Catalog can now be managed via KubermaticConfiguration through the field `.Spec.Applications.DefaultApplicationCatalog` ([#14697](https://github.com/kubermatic/kubermatic/pull/14697))

### Miscellaneous

> - A bug where images with digest instead of a tag are not properly parsed was fixed. This affected mirroring images and all parsing with an overwrite registry configured ([#14664](https://github.com/kubermatic/kubermatic/pull/14664))

Reviewer: Move this under Bugfixes

> - KubeLB: CCM will adjust the tenant kubeconfig to use API server endpoint and CA certificate from the management kubeconfig that is provided to KKP at the seed/datacenter level ([#14522](https://github.com/kubermatic/kubermatic/pull/14522))

Reviewer: Move this under Bugfixes

> - Support disabling default OperatingSystemProfiles in user clusters ([#14515](https://github.com/kubermatic/kubermatic/pull/14515))

Reviewer: Move this under New Features

> - The deprecated k8sgpt application has been removed and was replaced by the k8sgpt-operator app instead ([#14403](https://github.com/kubermatic/kubermatic/pull/14403))

Reviewer: Move this under Deprecations

> - Add a new optional field called `args` under KonnectivityProxySettings to allow users to specify a set of arguments for Konnectivity deployments ([#14189](https://github.com/kubermatic/kubermatic/pull/14189))

Reviewer: Move this under New Features

- Display system applications in cluster creation wizard and fix application type label for system applications ([#7388](https://github.com/kubermatic/dashboard/pull/7388))
- Make the Subnets field required when a VPC is selected, in both Wizard and Machine Deployment modes ([#7305](https://github.com/kubermatic/dashboard/pull/7305))
- Disable the Cluster Autoscaler option when the cluster autoscaler application is not defined in applications catalog ([#7283](https://github.com/kubermatic/dashboard/pull/7283))
- Add special characters restriction on Inputs and escape values to avoid rendering as HTML ([#7273](https://github.com/kubermatic/dashboard/pull/7273))
- Add role prioritization: Update logic to return the highest-priority role for members with multiple roles ([#7272](https://github.com/kubermatic/dashboard/pull/7272))
- Fix KKP login issue when the ID token is too large to be saved in a cookie, by splitting the token into multiple cookies ([#7206](https://github.com/kubermatic/dashboard/pull/7206))

#### Miscellaneous

> - Deprecate Equinix Metal provider ([#7376](https://github.com/kubermatic/dashboard/pull/7376))

Reviewer: Move this under Deprecations

> - Add new API endpoints for Kyverno integration ([#7106](https://github.com/kubermatic/dashboard/pull/7106))

Reviewer: Move this under New Features

> - Support infra storage classes and provider network subnets location compatibilities ([#7301](https://github.com/kubermatic/dashboard/pull/7301))

Reviewer: Move this under New Features

> - Dashboard has been upgraded to use Angular 19 ([#7183](https://github.com/kubermatic/dashboard/pull/7183))

Reviewer: Move this under New Features
